core/templates/components/state-editor/state-editor-properties-services/state-skill.service.ts

```diff
@@ -27,7 +27,10 @@ import { UtilsService } from 'services/utils.service';
 @Injectable({
   providedIn: 'root'
 })
-export class StateLinkedSkillIdService extends StatePropertyService<string> {
+export class StateLinkedSkillIdService extends
+  // Until a skill is selected, the state attribute is null. Also used to
+  // avoid circular dependencies.
+  StatePropertyService<string | null> {
```

Comment:
Null indicates this property(skills) not have been saved to the frontend yet. In StatePropertyService we have `displayed` property which play two roles
1. store save properties.
2. Working as boolean field as well, if we have not saved any properties through frontend we simply assign null/undefine to `displayed`.

So we can get rid of null. By using two different variable in StatePropertyService one `displayed` and another any boolean type variable.

But StatePropertyService is currently using in every state services through `extended` proporty. So we need to refactor almost all state property if we want to get rid of null completed.

And talking about this file we have circular dependencies. I tried to get rid of it but frontend is breaking. One PR get reverted because of this.